src/axom/primal/tests/primal_clip.cpp

```diff
@@ -1643,6 +1745,184 @@ TEST(primal_clip, tet_plane_intersect_four_edges)
+  // ON_POSITIVE_SIDE, ON_BOUNDARY, ON_POSITIVE in respect to the
+  // last edge of the clip polygon. Prior to fix, resulted in the
+  // vertex on the boundary being added twice.
+  {
+    Polygon2D subjectPolygon;
+    subjectPolygon.addVertex(Point2D {0.0, 0.0});
+    subjectPolygon.addVertex(Point2D {1.0, 0.0});
+    subjectPolygon.addVertex(Point2D {1.0, 1.0});
+
+    Polygon2D clipPolygon;
+    clipPolygon.addVertex(Point2D {0.0, 0.0});
+    clipPolygon.addVertex(Point2D {1.0, 0.0});
+    clipPolygon.addVertex(Point2D {0.0, 1.0});
+
+    Polygon2D poly = axom::primal::clip(subjectPolygon, clipPolygon, EPS);
+
+    EXPECT_NEAR(poly.signedArea(), 0.25, EPS);
+    EXPECT_EQ(poly.numVertices(), 3);
+  }
+
+  // Non-clipping - polygons intersect at a line
+  {
+    Polygon2D subjectPolygon;
+    subjectPolygon.addVertex(Point2D {0, 0});
+    subjectPolygon.addVertex(Point2D {1, 0});
+    subjectPolygon.addVertex(Point2D {1, 1});
+    subjectPolygon.addVertex(Point2D {0, 1});
+
+    Polygon2D clipPolygon;
+    clipPolygon.addVertex(Point2D {-1, 0});
+    clipPolygon.addVertex(Point2D {0, 0});
+    clipPolygon.addVertex(Point2D {0, 1});
+    clipPolygon.addVertex(Point2D {-1, 1});
+
+    Polygon2D poly =
+      axom::primal::clip(subjectPolygon, clipPolygon, EPS, CHECK_SIGN);
+
+    EXPECT_EQ(poly.isValid(), false);
+    EXPECT_EQ(poly.numVertices(), 0);
+  }
+
+  // Non-clipping - polygons intersect at a point
+  {
+    Polygon2D subjectPolygon;
+    subjectPolygon.addVertex(Point2D {0, 0});
+    subjectPolygon.addVertex(Point2D {1, 0});
+    subjectPolygon.addVertex(Point2D {1, 1});
+    subjectPolygon.addVertex(Point2D {0, 1});
+
+    Polygon2D clipPolygon;
+    clipPolygon.addVertex(Point2D {-1, -1});
+    clipPolygon.addVertex(Point2D {0, -1});
+    clipPolygon.addVertex(Point2D {0, 0});
+    clipPolygon.addVertex(Point2D {-1, 0});
+
+    Polygon2D poly =
+      axom::primal::clip(subjectPolygon, clipPolygon, EPS, CHECK_SIGN);
+
+    EXPECT_EQ(poly.isValid(), false);
+    EXPECT_EQ(poly.numVertices(), 0);
+  }
+
+  // Rosetta Code example
+  {
+    Polygon2D subjectPolygon;
+    subjectPolygon.addVertex(Point2D {50, 150});
+    subjectPolygon.addVertex(Point2D {200, 50});
+    subjectPolygon.addVertex(Point2D {350, 150});
+    subjectPolygon.addVertex(Point2D {350, 300});
+    subjectPolygon.addVertex(Point2D {250, 300});
+    subjectPolygon.addVertex(Point2D {200, 250});
+    subjectPolygon.addVertex(Point2D {150, 350});
+    subjectPolygon.addVertex(Point2D {100, 250});
+    subjectPolygon.addVertex(Point2D {100, 200});
+
+    Polygon2D clipPolygon;
+    clipPolygon.addVertex(Point2D {100, 100});
+    clipPolygon.addVertex(Point2D {300, 100});
+    clipPolygon.addVertex(Point2D {300, 300});
+    clipPolygon.addVertex(Point2D {100, 300});
+
+    Polygon2D poly = axom::primal::clip(subjectPolygon, clipPolygon, EPS);
+    EXPECT_NEAR(poly.signedArea(), 37083.3333333333, EPS);
+    EXPECT_EQ(poly.numVertices(), 10);
+  }
```

Comment:
👍 nice test! 

Looks like it's this polygon 
-- https://rosettacode.org/wiki/Sutherland-Hodgman_polygon_clipping#C++
-- https://static.miraheze.org/rosettacodewiki/0/0b/Sutherland-hodgman.png

If so, would it make sense to check that we get these vertices (or something close)?
```
(100, 116.667)
(125, 100)
(275, 100)
(300, 116.667)
(300, 300)
(250, 300)
(200, 250)
(175, 300)
(125, 300)
(100, 250)
```